Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -147,102 +191,211 @@ void FieldProbe::ComputeDiags (int step)
+
+                    /*
+                     * Determine whether or not to integrate field data.
+                     * If not integrating, store instantaneous values.
+                     */
+
+                    if (temp_field_probe_integrate != 1)
+                    {
+
+                        //Store Values on Particles
+
+                        part_Ex[ip] = Exp; //remember to add lorentz transform
+                        part_Ey[ip] = Eyp; //remember to add lorentz transform
+                        part_Ez[ip] = Ezp; //remember to add lorentz transform
+                        part_Bx[ip] = Bxp; //remember to add lorentz transform
+                        part_By[ip] = Byp; //remember to add lorentz transform
+                        part_Bz[ip] = Bzp; //remember to add lorentz transform
+                        part_S[ip] = S; //remember to add lorentz transform
+                    }
+                    else
+                    {
+                        part_Ex[ip] += Exp; //remember to add lorentz transform
+                        part_Ey[ip] += Eyp; //remember to add lorentz transform
+                        part_Ez[ip] += Ezp; //remember to add lorentz transform
+                        part_Bx[ip] += Bxp; //remember to add lorentz transform
+                        part_By[ip] += Byp; //remember to add lorentz transform
+                        part_Bz[ip] += Bzp; //remember to add lorentz transform
+                        part_S[ip] += S; //remember to add lorentz transform
+                    }
+                });// ParallelFor Close
+                if (field_probe_integrate == 0)
+                {
+                    for (int ip=0; ip < np; ip++)
+                    {
+                        // Fill output array
+                        m_data[0 * noutputs + ParticleVal::Ex] = part_Ex[ip];
+                        m_data[0 * noutputs + ParticleVal::Ey] = part_Ey[ip];
+                        m_data[0 * noutputs + ParticleVal::Ez] = part_Ez[ip];
+                        m_data[0 * noutputs + ParticleVal::Bx] = part_Bx[ip];
+                        m_data[0 * noutputs + ParticleVal::By] = part_By[ip];
+                        m_data[0 * noutputs + ParticleVal::Bz] = part_Bz[ip];
+                        m_data[0 * noutputs + ParticleVal::S] = part_S[ip];
+                    }
+                }
+                else
+                {
+                    amrex::Real const dt = WarpX::GetInstance().getdt(lev);
+                    static int iteration{0};
+                    if (m_intervals.contains(step+1))
+                    {
+                        amrex::Real time_ellapsed {iteration * dt};
+                        for (int ip=0; ip < np; ip++)
+                        {
+                            // Fill output array
+                            m_data[0 * noutputs + ParticleVal::Ex] = part_Ex[ip] * time_ellapsed;
```

Comment:
Requires managed memory without an explicit copy. Not a problem, but to be reworked later on.
```suggestion
                            // TODO replace with explicit copies later on to avoid managed memory on GPU
                            m_data[0 * noutputs + ParticleVal::Ex] = part_Ex[ip] * time_ellapsed;
```